gymnasium/utils/env_checker.py

```diff
@@ -165,6 +165,39 @@ def check_reset_options(env: gym.Env):
         )
 
 
+def check_step_return(env: gym.Env, seed=123):
```

Comment:
Could we change the name to `check_step_determinism`? We might want to change `check_reset_seed` to `check_reset_seed_determinism` to help make that clear as well